### PR TITLE
Fix getCurrentPackage

### PR DESCRIPTION
**Pull request by geof90**
When the native `getCurrentPackage` API rejects the promise, it does not return from the function, resulting in the `Callback with id xx: CodePush.getCurrentPackage() not found` errors reported in https://github.com/Microsoft/react-native-code-push/issues/195 and https://github.com/Microsoft/react-native-code-push/issues/182.

More investigation needs to be done to ascertain why the promise gets rejected in the first place, e.g. why the `app.json` file could go missing with the rest of the files in the update folder still remains intact.

This PR also modifies the `downloadUpdate` native APIs to clear away any existing files in a new update folder in order to prevent any chance of an update being corrupted. Also, on iOS, this changes `downloadUpdate` to use the `NSFileManager`'s `copyItemAtPath` method instead of the custom copy method when copying from an old update to a new one in the case of a diff update, with the assumption that the stock Objective-C API is more reliable.

### CODE REVIEW

File: android/app/src/main/java/com/microsoft/codepush/react/CodePush.java [hunk 1]
```diff
@@ -411,6 +411,10 @@ public void getCurrentPackage(final Promise promise) {
                 @Override
                 protected Void doInBackground(Object... params) {
                     WritableMap currentPackage = codePushPackage.getCurrentPackage();
+                    if (currentPackage == null) {
+                        promise.resolve("");
+                        return null;
```
Comment (lostintangent) [inline android/app/src/main/java/com/microsoft/codepush/react/CodePush.java]: Does this need to return null when the method has a void return type?

Comment (geof90) [inline android/app/src/main/java/com/microsoft/codepush/react/CodePush.java]: Its Void, not void. So yes

File: CodePushPackage.m [hunk 2]
```diff
@@ -259,9 +266,9 @@ + (void)downloadPackage:(NSDictionary *)updatePackage
                         return;
                     }
                     
-                    [CodePushPackage copyEntriesInFolder:currentPackageFolderPath
-                                              destFolder:newPackageFolderPath
-                                                   error:&error];
+                    [[NSFileManager defaultManager] copyItemAtPath:currentPackageFolderPath
+                                                            toPath:newPackageFolderPath
```
Comment (lostintangent) [inline CodePushPackage.m]: Does this create the directory if it doesn't exist?

File: CodePushPackage.m [hunk 3]
```diff
@@ -204,15 +204,22 @@ + (void)downloadPackage:(NSDictionary *)updatePackage
            failCallback:(void (^)(NSError *err))failCallback
 {
     NSString *newPackageFolderPath = [self getPackageFolderPath:updatePackage[@"packageHash"]];
-    NSError *error = nil;
+    NSError *error;
     
-    if (![[NSFileManager defaultManager] fileExistsAtPath:newPackageFolderPath]) {
-        [[NSFileManager defaultManager] createDirectoryAtPath:newPackageFolderPath
+    if (![[NSFileManager defaultManager] fileExistsAtPath:[self getCodePushPath]]) {
```
Comment (lostintangent) [inline CodePushPackage.m]: Why this change?

File: CodePushPackage.m (hunk 2 above)
Comment (geof90) [inline CodePushPackage.m]: It creates the destination folder if it doesnt exist, and fails if it already does

File: CodePushPackage.m (hunk 3 above)
Comment (geof90) [inline CodePushPackage.m]: Previously, we depended on the fact that we create the newPackageFolderPath if it doesnt exist to create the intermediate directories. Now, we only delete it if it exists, so we need this addition to create the intermediate directories, otherwise, the download handler (NSURLConnection) will fail

Comment (lostintangent) [inline CodePushPackage.m]: Would it make sense to make this an `else if` after the following clause?

File: android/app/src/main/java/com/microsoft/codepush/react/CodePush.java (hunk 1 above)
Comment (lostintangent) [inline android/app/src/main/java/com/microsoft/codepush/react/CodePush.java]: Ah OK, interesting. Another Java thing I didn't know about :)

File: CodePushPackage.m (hunk 3 above)
Comment (geof90) [inline CodePushPackage.m]: Well, yes. Though the expectation is that the `else if` only ever saves 1 file existence check during the first time this is called. Once the CodePush path is created, it is never removed unless a new binary is deployed.
